File: Algorithm/Leetcode/Code_Materials/python_source/charpter12_string/level2/reverse_k.py
# coding=utf-8
def reverseStr(s, k)
    if s is None or len(s) == 0:
        return s
    arr = list(s)
    n = len(arr)
    for i in range(0, n, 2 * k):
        arr[i:i+k] = reversed(arr[i:i+k])
    return "".join(arr)

s = "Hello World"
k = 3
print("Original String:", s)
reversed_str = reverseStr(s, k)
print("Reversed String:", reversed_str)

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter12_string/level2/reverse_k.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void reverse(char* arr, int left, int right) {
    while (left < right) {
        char temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
        left++;
        right--;
    }
}

char* reverseStr(char* s, int k) {
    if (s == NULL || strlen(s) == 0) {
        return s;
    }
    int n = strlen(s);
    for (int i = 0; i < n; i += 2 * k) {
        reverse(s, i, (i + k < n) ? i + k - 1 : n - 1);
    }
    return s;
}

int main() {
    char s[] = "Hello World";
    int k = 3;
    printf("Original String: %s\n", s);
    reverseStr(s, k);
    printf("Reversed String: %s\n", s);
    return 0;
}